sympy/integrals/integrals.py

```diff
@@ -153,6 +153,60 @@ def _eval_is_zero(self):
         if self.function.is_zero is False and got_none is False:
             return False
 
+    def is_divergent(self):
+        """
+        Check if the intergal is divergent for the integration limits.
+
+        Examples
+        ========
+
+        >>> from sympy import Integral, oo
+        >>> from sympy.abc import x
+        >>> f = x**2
+        >>> Integral(f,(x, 0, oo)).is_divergent()
+        True
+        >>> Integral(f,(x, )).is_divergent()
+        True
+        >>> f = x**3
+        >>> Integral(f,(x,oo)).is_divergent()
+        True
+        >>> f = x**3
+        >>> print(Integral(f,(x, )).is_divergent())
+        None
+        """
+        from sympy.sets import Interval
+        from sympy.solvers.inequalities import solve_univariate_inequality
+        from sympy.core.numbers import Rational
+
+        function = self.function
+        # check for divergent integrals
+        if len(self.limits) == 1:
+            xab = self.limits[0]
+            if xab[1:].free_symbols:
+                return None
+            if len(xab) == 3:
+                (x, a, b) = xab
+            elif len(xab) == 2:
+                x, a = xab[:2]
+                b = Rational(0)
+            else:
+                x = xab[0]
+                a, b = (-oo,oo)
+            xdom = Interval(Min(a,b),Max(a,b))
+            # check if one limit is oo and function limit at that is oo and
+            # check function non positive or non negative in integration limits
+            if any(i.is_infinite for i in (a,b)) and \
+                function.limit(x,[i for i in (a,b) if i.is_infinite][0]).is_infinite:
+                if solve_univariate_inequality(function>=0, x, domain=xdom, relational=False) == xdom:
```

Comment:
What if we only had this check and then if it passed, we could hand all the computation over to `Sum.is_convergent()` through the [integral test](https://en.wikipedia.org/wiki/Integral_test_for_convergence). We could do far more checks that way and avoid writing lots of code for `Integral.is_convergent()`.